WcaOnRails/spec/factories/persons.rb

```diff
@@ -3,9 +3,9 @@
   factory :person do
     wca_id do
       mid = ('A'..'Z').to_a.sample(4).join
-      "2016#{mid}01".tap do |id|
-        id.next! while Person.exists?(wca_id: id)
-      end
+      id = "2016#{mid}01"
+      id = id.next while Person.exists?(wca_id: id)
+      id
```

Comment:
I would personally ignore such an edge case in test code, but your 4 lines are at least as good as my four lines, so it doesn't matter.

I'm slightly frustrated that `foo = bar() while maybe(foo)` returns nil rather than `foo`. I suspect Ruby has a better construct like this, but I'm not spending any more time on this piece of bikeshed trivia this morning.